### PR TITLE
Adding simplification attribute to non-constructor-based equalities

### PR DESCRIPTION
**Pull request by traiansf**

### CODE REVIEW

File: kore/src/main/java/org/kframework/builtin/KLabels.java [hunk 1]
```diff
@@ -42,6 +42,7 @@
     public static final KLabel SetItem = KLabel("SetItem");
     public static final KLabel DotMap = KLabel(".Map");
     public static final KLabel List = KLabel("_List_");
+    public static final KLabel DotSet = KLabel(".Map");
```
Comment (daejunpark) [inline kore/src/main/java/org/kframework/builtin/KLabels.java]: `.Set` or `.Map`?

Comment (daejunpark) [reply kore/src/main/java/org/kframework/builtin/KLabels.java]: BTW, the order of these constants makes little sense.

Comment (traiansf) [reply kore/src/main/java/org/kframework/builtin/KLabels.java]: Thanks for catching this!  And abput the order, I agree, but it's not like it was very ordered before :-)

File: kore/src/main/scala/org/kframework/compile/ConstructorsChecks.scala [hunk 2]
```diff
@@ -0,0 +1,56 @@
+package org.kframework.compile
```
Comment (virgil-serbanuta) [inline kore/src/main/scala/org/kframework/compile/ConstructorsChecks.scala]: Some of the other files have copyright statements.

File: kore/src/main/scala/org/kframework/compile/ConstructorsChecks.scala [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    * Checks that a lhs pattern is constructor-based, i.e., an application of a
+    * function symbol to a list of constructor terms.
+    * @param m
+    * @param term
+    * @return
+    */
+  def constructorBased(m: Module, term: K): Boolean = term match {
+    case kapp: KApply => immutable(kapp.klist.items).forall(constructorOnly(m))
+    case t: KAs => constructorBased(m, t.pattern)
+    case other => true
```
Comment (virgil-serbanuta) [inline kore/src/main/scala/org/kframework/compile/ConstructorsChecks.scala]: Could you add a comment explaining why returning true here works?

**Review by virgil-serbanuta — COMMENTED** [review]
This looks reasonable to me, but I'd like someone with more kframework experience to approve.

**Review by dwightguth — CHANGES_REQUESTED** [review]
I'd really rather we didn't add to the amount of scala we are using in the project unnecessarily as it's already a source of problems on occasion

**Comment by traiansf** [comment]
@dwightguth what is your exact suggestion here?

**Comment by dwightguth** [comment]
I mean my suggestion is to write ConstructorChecks in Java.

**Comment by traiansf** [comment]
@dwightguth I've rewritten it in Java.  Please re-review.

**Review by dwightguth — APPROVED** [review]

**Pull request merged**